### PR TITLE
Add a notification when a runepouch degrades

### PR DESCRIPTION
**Pull request by Ishadijcks**
Implements #4089

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 1]
```diff
@@ -133,36 +138,41 @@ public void updateConfig(ConfigChanged event)
 	@Subscribe
 	public void onChatMessage(ChatMessage event)
 	{
-		if (event.getType() != ChatMessageType.SERVER || !config.showBindNeck())
+		if (event.getType() != ChatMessageType.SERVER && config.showBindNeck())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Not sure why this was changed in this way. At the very least, a guard clause for non-server messages should still remain:

```
if (event.getType() != ChatMessageType.SERVER)
{
    return;
}
```

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 2]
```diff
@@ -133,36 +138,41 @@ public void updateConfig(ConfigChanged event)
...
-				|| event.getMessage().contains("dust")
-				|| event.getMessage().contains("smoke")
-				|| event.getMessage().contains("mist"))
+			if (event.getMessage().contains("Your Binding necklace has disintegrated."))
 			{
-				bindNeckOverlay.bindingCharges -= 1;
-				return;
+				//set it to 17 because this message is triggered first before the above chat event
+				bindNeckOverlay.bindingCharges = 17;
 			}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: There should be a `return` statement after the above line.

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 3]
```diff
@@ -133,36 +138,41 @@ public void updateConfig(ConfigChanged event)
...
-
-		if (event.getMessage().contains("Your Binding necklace has disintegrated."))
+		if (config.degradingNotification())
 		{
-			//set it to 17 because this message is triggered first before the above chat event
-			bindNeckOverlay.bindingCharges = 17;
+			if (event.getMessage().contains("Your pouch has decayed through use."))
+			{
+				notifier.notify(POUCH_DECAYED_MESSAGE);
+			}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: There should be a `return` statement after the above line.

Comment (Ishadijcks) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Done

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 4]
```diff
@@ -166,13 +171,15 @@ public void onChatMessage(ChatMessage event)
 				//set it to 17 because this message is triggered first before the above chat event
 				bindNeckOverlay.bindingCharges = 17;
 			}
+			return;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: This should be within the above `if (match.find())` block, after the line where `bindNeckOverlay.bindingCharges` is set.

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 5]
```diff
@@ -166,13 +171,15 @@ public void onChatMessage(ChatMessage event)
...
 		}
 		if (config.degradingNotification())
 		{
 			if (event.getMessage().contains("Your pouch has decayed through use."))
 			{
 				notifier.notify(POUCH_DECAYED_MESSAGE);
 			}
+			return;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Same as other comment, this should be within the pouch decayed `if` block, after the `notifier.notify()` call.

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 6]
```diff
@@ -133,36 +138,48 @@ public void updateConfig(ConfigChanged event)
...
+				//set it to 17 because this message is triggered first before the above chat event
+				bindNeckOverlay.bindingCharges = 17;
+			}
+			return;
+		}
+		if (config.degradingNotification())
 		{
-			//set it to 17 because this message is triggered first before the above chat event
-			bindNeckOverlay.bindingCharges = 17;
+			if (event.getMessage().contains("Your pouch has decayed through use."))
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Use a constant for this string. This should probably be `POUCH_DECAYED_MESSAGE`, and the notification string should probably be something like `POUCH_DECAYED_NOTIFICATION_MESSAGE`.

Comment (Ishadijcks) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Fair enough, wasn't sure as it wasn't done with the binding necklace.

**Comment by Gamer1120** [comment]
Maybe it's an idea to say which pouch degraded? Like Large or Giant? I don't know whether that is worth the effort.

**Comment by Ishadijcks** [comment]
I'm not sure that is detectable. Also I think there is already a plugin that highlights broken pouches

**Comment by Gamer1120** [comment]
The degraded pouch changes color in inventory, so I'm pretty sure it's possible.

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 7]
```diff
@@ -133,36 +139,48 @@ public void updateConfig(ConfigChanged event)
...
+			if (event.getMessage().contains("Your Binding necklace has disintegrated."))
+			{
+				//set it to 17 because this message is triggered first before the above chat event
+				bindNeckOverlay.bindingCharges = 17;
 				return;
 			}
 		}
-
-		if (event.getMessage().contains("Your Binding necklace has disintegrated."))
+		if (config.degradingNotification())
```
Comment (NathenSample) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: `if(config.degradingNotification() && event.getMessage().contains(POUCH_DECAYED_MESSAGE))`

**Comment by NathenSample** [comment]
It is possible, this class already has logic which detects the presence of a broken pouch in your inventory, look at the logic around setting `degradedPouchInInventory` to true to see how it's done. Similar logic can be used to tell the user which of his pouches are decayed.

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java (hunk 7 above)
Comment (Ishadijcks) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Ah yeah I overlooked that during refactoring, good catch

**Pull request merged**